controllers/azurecluster_reconciler.go

```diff
@@ -138,7 +141,41 @@ func (r *azureClusterReconciler) Delete(ctx context.Context) error {
 			return errors.Wrapf(err, "failed to delete resource group")
 		}
 	}
+	if r.scope.AzureCluster.Spec.IdentityName != nil {
+		if err := r.deleteAzureIdentity(ctx); err != nil {
+			return errors.Wrapf(err, "failed to delete azure identity")
+		}
+	}
+
+	return nil
+}
 
+func (r *azureClusterReconciler) deleteAzureIdentity(ctx context.Context) error {
+	identityNamesToDelete := []string{}
+	var bindings aadpodv1.AzureIdentityBindingList
+	if err := r.scope.Client.List(ctx, &bindings, client.InNamespace("capz-system")); err != nil {
+		return errors.Wrapf(err, "failed to list AzureIdentityBinding")
+	}
+	for _, b := range bindings.Items {
+		binding := b
+		if binding.ObjectMeta.Labels[clusterv1.ClusterLabelName] == r.scope.AzureCluster.Name &&
+			binding.ObjectMeta.Labels[infrav1.ClusterLabelNamespace] == r.scope.AzureCluster.Namespace {
+
+			identityNamesToDelete = append(identityNamesToDelete, binding.Spec.AzureIdentity)
+			if err := r.scope.Client.Delete(ctx, &binding); err != nil {
+				return errors.Wrapf(err, "failed to delete AzureIdentityBinding")
+			}
+		}
+	}
+	for _, identityName := range identityNamesToDelete {
+		azureIdentity := &aadpodv1.AzureIdentity{}
+		if err := r.scope.Client.Get(ctx, client.ObjectKey{Name: identityName, Namespace: "capz-system"}, azureIdentity); err != nil {
```

Comment:
`capz-system` should be a const reused across the different places that have it hardcoded

Reply:
its hardcoded in the yaml, not in the code, I'm removing the hard-coding in the code, but don't know how to do that with yaml https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f9da6a5e82da5ccd7c692f47704015be27f7ba6e/config/default/kustomization.yaml#L1